### PR TITLE
Add an optional alias to RawKeyAccessor

### PR DESCRIPTION
**Pull request by AdamBJohnsonx**

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java [hunk 1]
```diff
@@ -60,21 +61,40 @@
...
+    /**
+     * The byte array that backs this key.
+     */
+    @NonNull
+    private final byte[] mKey;
+
+    /**
+     * The alias for the stored key, may be null.
+     */
+    private final String mAlias;
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java:83]: I don't see this ever being set

File: common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java [hunk 2]
```diff
@@ -193,7 +213,7 @@ public boolean verify(@NonNull final byte[] text,
     }
 
     @Override
-    @Nullable
+    @CheckForNull
```
Comment (shahzaibj) [inline common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java:216]: Why change this? How is this different from Nullable and what is the benefit?

Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java:216]: Well, it's not part of androidx, for one.  Lombok doesn't have a corresponding @Nullable, and this is the one settled on by the JEP.

File: common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java (hunk 1 above)
Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java:83]: Hm.  It wouldn't be, here.  I end up manufacturing these in the PRT code path, but I don't think we communicate a session key alias at that point.  I should set it in the random-key alias part of the KeyStoreAccessor, though.

File: common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java (hunk 2 above)
Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java:216]: Though it's been dormant for a while.  It depends on what we want to do with it.

File: common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java (hunk 1 above)
Comment (AdamBJohnsonx) [reply common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java:83]: 37ed32387b54c8540e6973e8c53ce52aa621eb72

File: common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java [hunk 3]
```diff
@@ -35,6 +33,7 @@
 import java.security.cert.Certificate;
 import java.util.Arrays;
 
+import javax.annotation.CheckForNull;
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java:36]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java (hunk 2 above)
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java:215]:
```suggestion
    @Nullable
```

File: common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java (hunk 3 above)
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java]:
```suggestion
import androidx.annotation.Nullablel;
```

File: common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java [hunk 4]
```diff
@@ -45,7 +43,9 @@
 import javax.crypto.spec.SecretKeySpec;
```
Comment (AdamBJohnsonx) [inline common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java:44]:
```suggestion
import androidx.annotation.Nullable;
```

File: common/build.gradle [hunk 5]
```diff
@@ -138,6 +138,8 @@ dependencies {
     implementation "com.google.code.gson:gson:$rootProject.ext.gsonVersion"
     implementation "androidx.browser:browser:$rootProject.ext.browserVersion"
     implementation "androidx.constraintlayout:constraintlayout:$rootProject.ext.constraintLayoutVersion"
+    implementation 'net.jcip:jcip-annotations:1.0'
+    implementation 'com.google.code.findbugs:jsr305:3.0.2'
```
Comment (AdamBJohnsonx) [inline common/build.gradle:142]:
```suggestion
```

File: common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java [hunk 6]
```diff
@@ -226,7 +244,7 @@ public SecureHardwareState getSecureHardwareState() {
 
     /**
      * Given this raw key, generate a derived key from it.  If we close on a KDF for hardware keys,
-     * this can get promoted to the symmetric key interface.
+     * this can get promoted to the symmetric key interface.  Derived keys have a null alias.
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/platform/RawKeyAccessor.java:247]: (mentioned offline, but putting here: let's doc why these are null aliased)

**Review by iambmelt — APPROVED** [review]

**Review by iambmelt — APPROVED** [review]
Signing off on latest iteration

**Pull request merged**